(base) PS C:\Users\User\github-classroom\Data_for_AI\dataforai\Data_for_AI> python -c "import nbformat; nb=nbformat.read('notebooks/01-data-profiling.ipynb', as_version=4); print(nb.cells[22].source)"

# ============================================
# HOTSPOT EXTRACTION
# ============================================

# Extract highest-risk pixels
hotspots = risk.sample(
    region=manhattan,
    scale=30,
    numPixels=10,
    geometries=True
)

# Convert to GeoJSON-style dictionary
hotspot_info = hotspots.getInfo()

print("Top hotspot samples extracted")
print(f"Number of hotspots: {len(hotspot_info['features'])}")
(base) PS C:\Users\User\github-classroom\Data_for_AI\dataforai\Data_for_AI> 

In [7]:
import json
import pandas as pd

with open("../data/profile-summary.json", "r") as f:
    profile = json.load(f)

print("Profile summary loaded successfully")

Profile summary loaded successfully


# Study Area and Data Sources

## Study Area

Manhattan, New York City

## Study Period

2018-06-01 to 2018-08-31

## Remote Sensing Sources

### Landsat 8 Collection 2 Level 2

Used for:

- Land Surface Temperature (LST)

Scenes processed:

- 2

### Sentinel-2 Surface Reflectance

Used for:

- Vegetation Density (NDVI)

Scenes processed:

- 7

In [8]:
study_info = pd.DataFrame({
    "Attribute": [
        "Study Area",
        "Time Period",
        "Landsat Scenes",
        "Sentinel Scenes"
    ],
    "Value": [
        profile["study_area"],
        profile["time_period"],
        profile["landsat_scene_count"],
        profile["sentinel_scene_count"]
    ]
})

study_info

,Attribute,Value
0,Study Area,Manhattan
1,Time Period,2018-06-01 to 2018-08-31
2,Landsat Scenes,2
3,Sentinel Scenes,7


# Land Surface Temperature Assessment

Land Surface Temperature (LST) was derived from Landsat 8 thermal imagery.

LST provides a measure of surface heating conditions and is widely used in urban heat island studies.

Higher values indicate greater thermal exposure.

In [9]:
lst = profile["lst_statistics"]

lst_table = pd.DataFrame({
    "Statistic": ["Minimum", "Mean", "Maximum"],
    "LST (°C)": [
        round(lst["LST_min"], 2),
        round(lst["LST_mean"], 2),
        round(lst["LST_max"], 2)
    ]
})

lst_table

,Statistic,LST (°C)
0,Minimum,10.33
1,Mean,30.73
2,Maximum,49.20


# Vegetation Density Assessment

Vegetation density was assessed using the Normalized Difference Vegetation Index (NDVI).

NDVI is a widely used indicator of vegetation coverage and environmental quality.

Vegetation can help reduce urban heat through shading and evapotranspiration.

In [10]:
ndvi = profile["ndvi_statistics"]

ndvi_table = pd.DataFrame({
    "Statistic": ["Minimum", "Mean", "Maximum"],
    "NDVI": [
        round(ndvi["NDVI_min"], 2),
        round(ndvi["NDVI_mean"], 2),
        round(ndvi["NDVI_max"], 2)
    ]
})

ndvi_table

,Statistic,NDVI
0,Minimum,-0.27
1,Mean,0.09
2,Maximum,0.70


# Heat Risk Framework

The project combines temperature and vegetation information into a single interpretable framework.

Areas with:

- Higher temperatures
- Lower vegetation density

are expected to exhibit greater relative heat exposure.

The framework uses normalized environmental indicators to calculate a heat risk score.

In [11]:
print("Heat Risk Score")
print()
print("Risk = Normalized LST - Normalized NDVI")

Heat Risk Score

Risk = Normalized LST - Normalized NDVI


# Heat Risk Logic Demonstration

The following examples illustrate how the framework behaves under different environmental conditions.

Higher temperatures increase risk.

Greater vegetation reduces risk.

In [12]:
examples = pd.DataFrame({
    "Scenario": [
        "Cool & Green",
        "Moderate Conditions",
        "Hot & Sparse Vegetation"
    ],
    "Normalized_LST": [
        0.20,
        0.60,
        0.95
    ],
    "Normalized_NDVI": [
        0.85,
        0.50,
        0.10
    ]
})

examples["Heat_Risk_Score"] = (
    examples["Normalized_LST"]
    - examples["Normalized_NDVI"]
)

examples

,Scenario,Normalized_LST,Normalized_NDVI,Heat_Risk_Score
0,Cool & Green,0.20,0.85,-0.65
1,Moderate Conditions,0.60,0.50,0.10
2,Hot & Sparse Vegetation,0.95,0.10,0.85


# Hotspot Identification

A heat risk layer was generated during the profiling workflow.

The highest-risk locations were extracted using Google Earth Engine sampling operations.

The extraction was performed using:

- 30 m sampling resolution
- Manhattan study boundary
- 10 hotspot samples

The resulting hotspot samples represent candidate locations for further investigation.

In [13]:
hotspot_summary = pd.DataFrame({
    "Metric": [
        "Hotspot Samples Extracted",
        "Sampling Resolution (m)"
    ],
    "Value": [
        10,
        30
    ]
})

hotspot_summary

,Metric,Value
0,Hotspot Samples Extracted,10
1,Sampling Resolution (m),30


# Interpretation

The environmental indicators reveal meaningful variation across Manhattan.

Observed conditions include:

### Land Surface Temperature

- Minimum: 10.33 °C
- Mean: 30.73 °C
- Maximum: 49.20 °C

### NDVI

- Minimum: -0.27
- Mean: 0.09
- Maximum: 0.70

The combination of these indicators supports identification of relative urban heat hotspots.

# Limitations

## Temporal Scope

The analysis represents Summer 2018 conditions only.

## Relative Risk Interpretation

The heat risk score represents relative heat exposure rather than direct measurements of human thermal comfort or health risk.

## External Validation

Comparison with the NYC Heat Vulnerability Index remains future work.

## Spatial Resolution

Results depend on the spatial resolution of Landsat 8 and Sentinel-2 imagery.

# Session 4 Outcome

Session 4 produced a reproducible urban heat hotspot identification workflow based on:

- Landsat 8 Land Surface Temperature
- Sentinel-2 Vegetation Density
- Heat Risk Scoring
- Hotspot Identification

Produced Outputs:

- Land Surface Temperature layer
- NDVI layer
- Heat Risk layer
- Hotspot extraction results
- Profile summary statistics

The workflow establishes the analytical foundation for Session 5 evaluation and future decision-support tool development.